pkg/api/ui_handler.go

```diff
@@ -61,8 +61,29 @@ func NoCacheHandler(handler http.Handler) http.Handler {
 	})
 }
 
+// BaseURLHandler ensures that routes under BaseURL (i.e. /api/v1) are returned
+// with ErrInvalidAPIEndpoint.
+func BaseURLHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Print("printing the url.path now", r.URL.Path)
+		if r.URL.Path != "/" {
```

Comment:
The server uses mux to handle the requested route, if we register the invalid API error handler on the base URL route - it will catch all the calls that don't match the current API handlers.
This way we will not require any 'if' to check the 'r.URL.Path' because the router will do this job for us.

```go
r.Mount(BaseURL, http.HandleFunc(InvalidAPIEndpointHandler))
```

This handler will be just response with an error:

```go
func InvalidAPIEndpointHandler(w http.ResponseWriter, _ *http.Request) {
	writeError(w, http.StatusInternalServerError, ErrInvalidAPIEndpoint)
}
```

With the above, we can move this code to `serve.go` - no longer related to the UI handler.

Reply:
Done